packages/protocol-kit/src/contracts/utils.ts

```diff
@@ -152,6 +152,14 @@ const memoizedGetProxyCreationCode = createMemoizedFunction(
   }
 )
 
+/**
+ * Provides a default salt nonce with appended chainId to generate different addresses for the
+ * same Safe in different networks.
+ */
+export function getChainSpecificDefaultSaltNonce(chainId: number): string {
```

Comment:
What is your opinion about to provide a full JSDoc for this function?. Here's a suggested format:

```javascript
/**
 * Provides a chain-specific default salt nonce for generating unique addresses 
 * for the same Safe configuration across different chains.
 * 
 * @param {number} chainId - The chain ID associated with the chain.
 * @returns {string} The chain-specific salt nonce in hexadecimal format.
 */
```

Reply:
Yes makes total sense! Added it

Reply:
🤓 : the largest number in javascript is (2^53) – 1, but chain IDs in ethereum theoretically can be 256 bits numbers

Reply:
From [here](https://eips.ethereum.org/EIPS/eip-1344): 
> Adds a new opcode CHAINID at 0x46, which uses 0 stack arguments. It pushes the current chain ID onto the stack. Chain ID is a 256-bit value. The operation costs G_base to execute.

Reply:
Ah ok, so `BigInt` should be used? But would it not be better then to change it everywhere in the repo, instead of only in this function? Because the `ethAdapter.getChainId()` returns a simple `number` value and that is where the chainId comes from when being passed to this function.

Reply:
Ah, then it is irrelevant for this PR. Thanks for the explanation

Reply:
Makes sense! I will update the type to BigInt in the PR of the Ethers v6 migration.

Reply:
BigInt or string both should work. Ethers also has a `BigNumberish` type

Reply:
I think that now we have BigInt available in Javascript is the way to go. Also most libraries I expect they will use BigInt as for example Ethers v6 already does.

We created a separate ticket to handle this as there are quite some changes needed. Thank you for your input @mmv08 

https://github.com/safe-global/safe-core-sdk/issues/564